src/context/AuthContext.tsx

```diff
@@ -1,7 +1,51 @@
-import React from 'react'
 
-export default function AuthContext() {
-  return (
-    <div>AuthContext</div>
-  )
+import  { jwtDecode} from "jwt-decode";
+import { ReactNode, createContext, useContext, useEffect, useState } from "react";
+
+interface AuthContextType {
+  saveAdminData: () => void;
+  adminData?: string | null;
 }
+
+export const AuthContext = createContext<AuthContextType | null>(null);
+
+export const AuthContextProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const [adminData, setAdminData] = useState<string | null>(null);
+
+  const saveAdminData = () => {
+    const encodedToken = localStorage.getItem("adminToken");
+    if (encodedToken) {
+      const decodedToken = jwtDecode(encodedToken) as string;
+      setAdminData(decodedToken);
+    }
+  };
+
+  useEffect(() => {
+    saveAdminData();
+  }, []);
+
+  const contextValue: AuthContextType = {
+    adminData,
+    saveAdminData
+  };
+
+  return (
+    <AuthContext.Provider value={contextValue}>
+      {children}
+    </AuthContext.Provider>
+  );
+};
+
+
+// Custom hook to use the toast context
+export const useToken = (): AuthContextType => {
```

Comment:
The name `useToken` for the custom hook suggests that we will return the token and maybe a function that will be repsonsible for updating the token. I suggest renaming it to something such as `useAuthContext` or `useUser`